### DISCUSSION TITLE
Updating Dots

### DISCUSSION THREAD

**Discussion by Enigma1309** [prasanthrangan/hyprdots#950] [General]
Hey there, i really like this repo and it's development, but there are few things which i wanna keep on my own say kitty.conf, so is there any way i can i can enjoy the latest development features without losing my personal config
i've seen #262 and i know we can get our files back from the backup directory, but what's the point of updating when we are restoring the same config ?
as far as i've been using his repo, each commit brings some new and better functionality but i can't find a way to keep my config up to date, i manually used to see the commits and using git diff i use to make new files and features which i wanna use,
is there any way i can keep my configs and still updates the parts i wanna do, using ./install.sh -r copies the whole dots to the sysytem, can we implement something like diff base update so when user updates i shows in the diff of the current version and the new version and manually resolve the merge, ik this will be hectic but it would be worth it
also if anyone have any other suggestion please share

**Comment by kRHYME7** [comment]
If you really want to have all those micro changes example line by line, then use git to manage your own fork.  

BUT if managing files  you can use this ``` Hyprdots update ``` then  ``` Hyprdots upgrade ``` to manage files. 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/60285d3a-7213-4899-822c-b43c567e155c)



Here is my configuration list ``` ~/.config/hyprdots-ctl/manage_cfg.lst ``` 
With this I can keep all custom files.
```
P|${HOME}|.zshrc .p10k.zsh|zsh oh-my-zsh-git zsh-theme-powerlevel10k pokemon-colorscripts-git
P|${HOME}/.config/Code/User|settings.json|visual-studio-code-bin
S|${HOME}/.config|dunst|dunst
S|${HOME}/.config/hypr|animations.conf hyprland.conf windowrules.conf scripts themes wallbash|hyprland
P|${HOME}/.config/hypr|theme.ctl|hyprland
P|${HOME}/.config/hypr|keybindings.conf monitors.conf userprefs.conf|hyprland
P|${HOME}/.config/hypr|nvidia.conf|hyprland nvidia-utils
P|${HOME}/.config|kitty|kitty
S|${HOME}/.config|Kvantum|kvantum
S|${HOME}/.config|MangoHud|mangohud
S|${HOME}/.config|neofetch|neofetch
S|${HOME}/.config|qt5ct|qt5ct
S|${HOME}/.config|rofi|rofi
S|${HOME}/.config|swaylock|swaylock-effects
S|${HOME}/.config|swww|swww
S|${HOME}/.config|config.jsonc style.css modules|waybar
P|${HOME}/.config|config.ctl|waybar
S|${HOME}/.config|wlogout|wlogout
S|${HOME}/.config|nwg-look gtk-3.0 xsettingsd|nwg-look
S|${HOME}/.icons|default|nwg-look
S|${HOME}|.gtkrc-2.0|nwg-look
P|${HOME}/.config|dolphinrc kdeglobals|dolphin
P|${HOME}/.local/share|dolphin|dolphin
P|${HOME}/.local/share/kxmlgui5|dolphin|dolphin
S|${HOME}/.config|lsd|lsd
S|${HOME}/.config|xsettingsd nwg-look gtk-3.0|nwg-look
```

**Reply by kRHYME7** [reply]
I discussed this here : 
https://github.com/prasanthrangan/hyprdots/discussions/863#discussioncomment-8439994

**Reply by Enigma1309** [reply]
Let me test this..

**Reply by kRHYME7** [reply]
For waybar create your on module then just don't let the config.ctl for be overwritten,  you can follow my configuration list. I'm testing this for almost a month.

**Reply by Enigma1309** [reply]
can you explain the syntax better 
i understand O,PS,I but i don't get how the directory structure is being managed why are we using pipe "|"?

**Reply by Enigma1309** [reply]
also is there a way i can blacklist sddm or any other software i don't wanna install?

**Reply by kRHYME7** [reply]
>  i understand O,PS,I but i don't get how the directory structure is being managed why are we using pipe "|"?
 
 
 Those are just columns, that's the convention use in Hyprdots so I adapted it. 
 
 
1st columns is the FLAG
2nd col. is the file or directory  path 
3rd column is the package dependency (incase you don't have the package )

 

> also is there a way i can blacklist sddm or any other software i don't wanna install?


Navigate to your clone directory > Scripts/ 
``` $HOME/Hyprdots/Scripts/ ```

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/22931502-a4c1-4eee-b843-f372035df323)

remove the sddm line.

**Reply by Enigma1309** [reply]
Thank you it worked

**Comment by Enigma1309** [comment]
Where does the `Hyprdots update` stores github repo, is it in Home/Hyprdots??

**Reply by kRHYME7** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/716f999a-5233-4775-80e9-541f4f085c0b)

It is the Clone Directory on which you run ``` git clone ```  upon installation of Hyprdots

In my case I clone it somewhere. Normally it is in ``` /home/username/Hyprdots ```

**Reply by Enigma1309** [reply]
thanks i got sddm removed from the app list and updated it
